libs/sdk-ui-kit/README-DEV.MD

```diff
@@ -0,0 +1,275 @@
+    &--style {
+        &-primary {
+            // styles for primary variant
+
+            &#{$root}--isSelected {
+                // combined styles for primary and isSelected
+            }
+        }
+    
+        &-secondary {
+            // styles for secondary variant
+
+            &#{$root}--isSelected {
+                // combined styles for secondary and isSelected
+            }
+        }
+    }
+}
+```
+
+### Nested Element Styles
+
+```tsx
+const { b, e } = bem("gd-ui-kit-button");
+
+<button className={b()}>
+    <span className={e("icon")} />
+</button>
+```
+
+```scss
+.gd-ui-kit-button {
+    // component styles
+    
+    &__icon {
+        // styles for nested element
+    }
+}
+```
+
+### Nested Element Styles Based on Props
+
+```tsx
+interface IProps {
+    isSelected: boolean;
+    style: "primary" | "secondary",
+}
+
+const { b, e } = bem("gd-ui-kit-button");
+
+<button className={b()}>
+    <span className={e("icon", { isSelected, style })} />
+</button>
+```
+
+```scss
+.gd-ui-kit-button {
+    // component styles
+
+    &__icon {
+        // styles for nested element
+
+        &--isSelected {
+            // styles when isSelected is true
+        }
+
+        &--style {
+            &-primary {
+                // styles for primary variant
+            }
+        
+            &-secondary {
+                // styles for secondary variant
+            }
+        }
+    }
+}
+```
+
+### Nested Element Style Overrides
+*Note: Avoid overrides when possible, as they reduce readability and maintainability. Prefer top-level derived styles, but use overrides if necessary.*
+
+```tsx
+interface IProps {
+    isSelected: boolean;
+}
+
+const { b, e } = bem("gd-ui-kit-button");
+
+<button className={b({ isSelected })}>
+    <span className={e("icon")} />
+</button>
+```
+
+```scss
+.gd-ui-kit-button {
+    // component styles
+    $root: &;
+
+    &__icon {
+        // styles for nested element
+    }
+
+    &--isSelected {
+         &#{$root}__icon {
+            // style overrides for icon when isSelected is true
+        }
+    }
+
+    &:active {
+        &#{$root}__icon {
+            // style overrides for icon when active
+        }
+    }
+    
+}
+```
+
+## Theming
+
+All components are respecting our theming.
+
+CSS variables (CSS properties e.g. --gd-custom-property) to use should be included in Figma designs, if you are not sure which one is correct, discuss it with designers.
+
+All default theme css properties values can be found in [defaultTheme.scss](../sdk-ui-kit/src/@ui/defaultTheme.scss).
+
+If you are adding a new CSS property, you should add its specification [here](../sdk-ui-theme-provider/src/variablesSpec/allThemeCssVariables.ts), to its respective location.
+This is required, because these specifications are used to generate default theme, and for validation of the consistent properties usage to keep our theming consistent.
+
+
+## Layouting
+
+Component should care only about its own layout.
```

Comment:
Must here and the next sentence.

Reply:
Updated. I also added new readme for more details about theming (see the last commit).